alluxio/alluxio.sh

```diff
@@ -0,0 +1,173 @@
+
+SPARK_HOME=${SPARK_HOME:-"/usr/lib/spark"}
+HIVE_HOME=${HIVE_HOME:-"/usr/lib/hive"}
+HADOOP_HOME=${HADOOP_HOME:-"/usr/lib/hadoop"}
+
+# Script constants
+ALLUXIO_VERSION=2.0.1
+ALLUXIO_HOME=/opt/alluxio
+ALLUXIO_SITE_PROPERTIES=${ALLUXIO_HOME}/conf/alluxio-site.properties
+ALLUXIO_DOWNLOAD_URL=https://downloads.alluxio.io/downloads/files/${ALLUXIO_VERSION}/alluxio-${ALLUXIO_VERSION}-bin.tar.gz
+
+# Downloads a file to the local machine from a remote HTTP(S) or GCS URI into the cwd
+#
+# Args:
+#   $1: URI - the remote location to retrieve the file from
+download_file() {
+  local uri="$1"
+
+  if [[ "${uri}" == gs://* ]]
+  then
+    sudo gsutil cp "${uri}" ./
+  else
+    # TODO Add metadata header tag to the wget for filtering out in download metrics.
+    sudo wget -nv "${uri}"
+  fi
+}
+
+# Appends a property KV pair to the alluxio-site.properties file
+#
+# Args:
+#   $1: property name
+#   $2: property value
+append_alluxio_property() {
+  local property="$1"
+  local value="$2"
+
+  echo "${property}=${value}" >> ${ALLUXIO_SITE_PROPERTIES}
+}
+
+# Calculates the default memory size as 1/3 of the total system memory
+#
+# Echo's the result to stdout. To store the return value in a variable use
+# val=$(get_defaultmem_size)
+get_default_mem_size() {
+  local mem_div=3
+  local phy_total=$(free -m | grep -oP '\d+' | head -n1)
+  local mem_size=$(( ${phy_total} / ${mem_div} ))
+  echo "${mem_size}MB"
+}
+
+# Download the Alluxio tarball and untar to ALLUXIO_HOME
+function bootstrap_alluxio() {
+  mkdir ${ALLUXIO_HOME}
+  download_file ${ALLUXIO_DOWNLOAD_URL}
+  local tarball_name=${ALLUXIO_DOWNLOAD_URL##*/}
+  sudo tar -zxf ${tarball_name} -C ${ALLUXIO_HOME} --strip-components 1
+  sudo ln -s ${ALLUXIO_HOME}/client/*client.jar ${ALLUXIO_HOME}/client/alluxio-client.jar
+
+  # Download files to /opt/alluxio/conf
+  local download_files_list=$(/usr/share/google/get_metadata_value attributes/download_files_list)
+  local download_delimiter=";"
+  IFS="${download_delimiter}" read -ra files_to_be_downloaded <<< "${download_files_list}"
+  if [ "${#files_to_be_downloaded[@]}" -gt "0" ]; then
+    for file in "${files_to_be_downloaded[@]}"; do
+      local filename="$(basename ${file})"
+      download_file "${file}"
+      sudo mv "${filename}" "${ALLUXIO_HOME}/conf/${filename}"
+    done
+  fi
+
+  # Configure client applications
+  sudo mkdir -p ${SPARK_HOME}/jars/
+  sudo ln -s "${ALLUXIO_HOME}/client/alluxio-client.jar" ${SPARK_HOME}/jars/alluxio-client.jar
+  sudo mkdir -p ${HIVE_HOME}/lib/
+  sudo ln -s "${ALLUXIO_HOME}/client/alluxio-client.jar" ${HIVE_HOME}/lib/alluxio-client.jar
+  sudo mkdir -p ${HADOOP_HOME}/lib/
+  sudo ln -s "${ALLUXIO_HOME}/client/alluxio-client.jar" ${HADOOP_HOME}/lib/alluxio-client.jar
+  sudo systemctl restart hive-metastore
+  sudo systemctl restart hive-server2
+
+  # Optionally configure presto
+  # OK to fail in this section
+  set +o errexit
+  PRESTO_HOME=${PRESTO_HOME:-$(ls -d -- /presto-server*)}
```

Comment:
This is not going to work with https://github.com/GoogleCloudPlatform/dataproc-initialization-actions/blob/79db60ee3b2cc302a50d6d9202398eee000fdeb9/starburst-presto/presto.sh#L53-L54
PRESTO_HOME will be set to `/presto-server-312... /presto-server`. 

- either use `/presto-server-*` here
- or we need to introduce a convention that Presto is installed in `/presto-server` directory (without version suffix).

@medb how do you usually solve problems like this?

Reply:
thx @findepi. i see the issue (`ls` matches 2 directory names). a possible workaround would be to choose the first one, but your previous comment is still valid. the order of init action execution is counterintuitive. not sure how to handle this other than modifying all flavors of presto to create the symlink as starburst presto does.

Reply:
> modifying all flavors of presto to create the symlink as starburst presto does.

Unfortunately this doesn't solve the problem either. If your init script is run before Starburst Presto init script, yours will create `/presto-server` _directory_ (`mkdir -p /presto-server/plugin/hive-hadoop2`) and Starburst Presto init script will _fail_ to create `/presto-server` _symlink_.
(Yes, as it is, your init script cannot be used together with Starburst Presto init script, regardless of order.)

A possible approach would be to strip version number from `/presto-server-{version}`. Ie Presto init action (both init currently existing) could create `/presto-server` _directory_, gracefully handling it if the directory already exist. This solves "plug the jar problem". You still need to know whether to restart Presto or not, since Presto inits classpath based on existing jar, a new jar will not be auto-added to plugin's classpath.

Anyway, treat the above as technically possible idea only. For example, I don't know if the version in `/presto-server-{version}` directory name is used/useful for something else.
I'd love to hear from @medb or someone else from Dataproc team what's the "by the book" approach to this.

Reply:
Usually we use `compgen -G /presto-server-*` to resolve glob in paths, it returns first matched path, so it will work in this case.

But, I think that canonical approach is to have non-versioned install folders, so installation dir is the same across all presto versions.

Reply:
I have changed init action to expect presto only in the `/opt/presto-server` directory - I will work to modifying Presto init actions to link presto installation there.